wcfsetup/install/files/lib/system/mail/Mail.class.php

```diff
@@ -327,10 +221,26 @@ public function getMessage() {
 	 */
 	public function setFrom($from) {
 		if (is_array($from)) {
-			$this->from = self::buildAddress(key($from), current($from));
+			$this->email->setSender(new Mailbox(current($from), key($from)));
+		}
+		else {
+			$this->email->setSender(new Mailbox($from));
+		}
+	}
+	
+	/**
+	 * Adds recipients as to, cc or bcc to this mail.
+	 *
+	 * @param	mixed		$recipient
+	 * @param	string		$type
+	 * @since 3.2.0
+	 */
+	protected function addRecipient($recipient, $type = 'to') {
+		if (is_array($recipient)) {
+			$this->email->addRecipient(new Mailbox(current($recipient), key($recipient), $this->language), $type);
 		}
 		else {
-			$this->from = $from;
+			$this->email->addRecipient(new Mailbox($recipient, null, $this->language), $type);
```

Comment:
This change is bogus. `$recipient` might be an address specification in `name-addr` format.